lua/scriptwrapper.lua

```diff
@@ -26,27 +28,62 @@ local TryRunSandboxed = function(fn)
 	end
 end
 
+local function merge(dest, source)
+	for k,v in pairs(source) do
+		dest[k] = dest[k] or v
+	end
+	return dest
+end
+
+local TryRun = function(fn)
+	local env = merge(environment, _G)
```

Comment:
I didn't understand what this was doign at a first glance, and even after a first pass of digging/testing I still don't understand why directly setting `setfenv(environment)` isn't sufficient.

Can you please add a comment explaining why this is needed / what this is doing?

Reply:
It was months ago that I wrote this, but I think what it does is merging the OpenRA Lua API with an unrestricted system Lua table which is not supported by Lua itself therefore this helper method.